packages/wmr/src/plugins/wmr/client.js

```diff
@@ -228,3 +236,173 @@ function updateStyleSheet(url) {
+		:root {
+			--bg: #fff;
+			--bg-code-frame: rgb(255, 0, 32, 0.1);
+			--bg-active-line: #fbcecc;
+			--text: #222;
+			--text2: #444;
+			--title: #e84644;
+			--code: #333;
+			font-family: sans-serif;
+			line-height: 1.4;
+			color: var(--text);
+			background: var(--bg);
+		}
+
+		* {
+			box-sizing: border-box;
+		}
+		
+		@media (prefers-color-scheme: dark) {
+			:root {
+				--bg-code-frame: rgba(251, 93, 113, 0.2);
+				--bg-active-line: #4f1919;
+				--bg: #353535;
+				--text: #f7f7f7;
+				--text2: #ddd;
+				--code: #fdd1d1;
+			}
+		}
+
+		.close {
+			cursor: pointer;
+			position: absolute;
+			top: 1rem;
+			right: 1rem;
+		}
+
+		.inner {
+			max-width: 80ch;
+			padding: 4rem 1rem;
+			margin: 0 auto;
+		}
+
+		.title {
+			color: var(--title);
+			font-weight: normal;
+			font-size: 1.5rem;
+		}
+
+		.code-frame {
+			overflow: auto;
+			padding: 0.5rem;
+			background: var(--bg-code-frame);
+			color: var(--code);
+		}
+		.line {
+			padding: 0.25rem 0.5rem;
+		}
+		.active-line {
+			display: inline-block;
+			width: 100%;
+			background: var(--bg-active-line);
+		}
+
+		.detail {
+			cursor: pointer;
+			color: var(--text2);
+		}
+		.stack-frame {
+			padding: 0.5rem 0;
+		}
+		.stack-name {
+			color: var(--text);
+		}
+		.stack-loc {
+			color: var(--text2);
+			font-family: monospace;
+		}
+	`;
+		const overlay = createDom('div', {});
+		const close = createDom('button', {
+			class: 'close',
+			onclick: () => {
+				window.postMessage('wmr-close-error-overlay', '*');
+			}
+		});
+		close.textContent = 'close';
+		const inner = createDom('div', { class: 'inner' });
+		overlay.append(close, inner);
+
+		const title = createDom('h1', { class: 'title' });
+		title.textContent = data.error;
+		inner.append(title);
+
+		const codeFrame = createDom('pre', { class: 'code-frame' });
+		const code = createDom('code', {});
+		codeFrame.append(code);
+
+		data.codeFrame.split('\n').forEach((line, i, arr) => {
+			const dom = createDom('span', {
+				class: 'line' + (line.startsWith('>') ? ' active-line' : '')
+			});
+			dom.textContent = line;
+			code.append(dom, i < arr.length - 1 ? '\n' : '');
+		});
+		inner.append(codeFrame);
+
+		outer.append(style, overlay);
+
+		const stackDetail = createDom('details', { class: 'detail' });
+		const stackSummary = createDom('summary', {});
+		stackSummary.textContent = `${data.stack.length} stack frames were collapsed.`;
+		stackDetail.append(stackSummary);
+
+		const frames = data.stack.map(frame => {
+			const container = createDom('div', { class: 'stack-frame' });
+			const name = createDom('div', { class: 'stack-name' });
+			name.textContent = frame.name;
+			const loc = createDom('div', { class: 'stack-loc' });
+			loc.textContent = `${frame.fileName}:${frame.line}:${frame.column}`;
+
+			container.append(name, loc);
+			return container;
+		});
+		stackDetail.append(...frames);
+
+		inner.append(stackDetail);
+		iframe.contentDocument?.body.appendChild(outer);
```

Comment:
@marvinhagemeister I took a swing at making this use a more hyperscript-style `h()` that supports children and properties:

```suggestion
		const doc = iframe.contentDocument;

		function h(tag, props, ...children) {
			props = props || {};
			tag = tag.replace(/([.#])([^.#]+)/g, (s,g,i) => ((props[g=='.'?'className':'id']=i), ''));
			const el = Object.assign(doc.createElement(tag), props);
			el.append(...children);
			return el;
		}

		const STYLE = `
		:root {
			--bg: #fff;
			--bg-code-frame: rgb(255, 0, 32, 0.1);
			--bg-active-line: #fbcecc;
			--text: #222;
			--text2: #444;
			--title: #e84644;
			--code: #333;
			font-family: sans-serif;
			line-height: 1.4;
			color: var(--text);
			background: var(--bg);
		}

		* {
			box-sizing: border-box;
		}

		@media (prefers-color-scheme: dark) {
			:root {
				--bg-code-frame: rgba(251, 93, 113, 0.2);
				--bg-active-line: #4f1919;
				--bg: #353535;
				--text: #f7f7f7;
				--text2: #ddd;
				--code: #fdd1d1;
			}
		}

		.close {
			cursor: pointer;
			position: absolute;
			top: 1rem;
			right: 1rem;
		}

		.inner {
			max-width: 80ch;
			padding: 4rem 1rem;
			margin: 0 auto;
		}

		.title {
			color: var(--title);
			font-weight: normal;
			font-size: 1.5rem;
		}

		.code-frame {
			overflow: auto;
			padding: 0.5rem;
			background: var(--bg-code-frame);
			color: var(--code);
		}
		.line {
			padding: 0.25rem 0.5rem;
		}
		.active-line {
			display: inline-block;
			width: 100%;
			background: var(--bg-active-line);
		}

		.detail {
			cursor: pointer;
			color: var(--text2);
		}
		.stack-frame {
			padding: 0.5rem 0;
		}
		.stack-name {
			color: var(--text);
		}
		.stack-loc {
			color: var(--text2);
			font-family: monospace;
		}
		`;

		const lines = data.codeFrame.split('\n').reduce((lines, line, i, arr) => {
			lines.push(h('span', {
				className: 'line' + (line.startsWith('>') ? ' active-line' : '')
			}, line));
			if (i < arr.length - 1) lines.push('\n');
			return lines;
		}, []);

		const frames = data.stack.map(frame =>
			h('div.stack-frame', null,
				h('div.stack-name', null, frame.name),
				h('div.stack-loc', null, `${frame.fileName}:${frame.line}:${frame.column}`)
			)
		);

		doc.body.append(
			h('div#wmr-error-overlay', null,
				h('style', null, STYLE),
				h('div', null, 
					h('button.close', {
						onclick() {
							errorOverlay.remove();
							errorOverlay = null;
						},
					}, 'close'),
					h('div.inner', null,
						h('h1.title', null, String(data.error)),
						h('pre.code-frame', null, h('code', null, lines)),
						h('details.detail', null,
							h('summary', null, `${data.stack.length} stack frames were collapsed.`),
							...frames
						)
					)
				)
			)
		);
```